scripts/build_android.sh

```diff
@@ -114,7 +116,7 @@ if [ -z "$MAX_JOBS" ]; then
     MAX_JOBS=$(nproc)
   fi
 fi
-cmake --build . -- "-j${MAX_JOBS}"
+cmake --build . --target install -- "-j${MAX_JOBS}"
```

Comment:
IIUC what this does is also install the libraries into whatever CMAKE_INSTALL_PREFIX is set to, /usr/local by default. Why do you want this behavior? This should be an exact copy of $INSTALL_PREFIX == $CAFFE2_ROOT/build_android/install

Reply:
Instead of this, we can also set $INSTALL_PREFIX inside a 
```
if [[ -z "$INSTALL_PREFIX" ]]; then
    INSTALL_PREFIX=${BUILD_ROOT}/install
fi
```
And add some documentation saying that to set INSTALL_PREFIX when you want to install. You'd then get this behavior with
```
INSTALL_PREFIX="${CMAKE_INSTALL_PREFIX:-/usr/local}" build_android.sh ...
```

Reply:
We already do pass `-DCMAKE_INSTALL_PREFIX=$INSTALL_PREFIX` in the first invocation of cmake.
Without the `--target install`, I don't get anything installed in the install prefix though.

Reply:
Won't this line install things https://github.com/pytorch/pytorch/blob/master/scripts/build_android.sh#L121 ?

Reply:
Surprisingly, perhaps, no. (I spent quite a bit of time poking at this, even though it wasn't the first time. It also was a main reason I initially adapted PyTorch's build mechanism.)

Reply:
You're just trying to install header files into /usr/local along with the other libraries produced by build_android.sh right? You're not concerned about any Python installations?

All the Pytorch build mechanism does is call 'make install' anyway https://github.com/pytorch/pytorch/blob/master/tools/build_pytorch_libs.py#L281 , so this should be no different. Pytorch's build system is different only because it's wraps everything in setup.py, which also copies a bunch of things into the active Python installation. 

But, I don't think this breaks anything. So if you need this for it to work then I'm okay with it.

Reply:
I might just doing be something stupid.
Without this PR, are you able to get cmake to install the headers in `build_android/install/`? What is your call scripts/build_android.sh?

Reply:
Do you happen to have ninja installed? I just asked David Reiss about this and realized that we should do something like 
```
https://github.com/pytorch/pytorch/blob/master/tools/build_pytorch_libs.py#L284-L291
```
and use ninja if that's installed instead. So actually, let's keep this change and just delete the ```make install``` line below.

Reply:
Ah, cool, thanks for following up.